Sources/Commands/SwiftTestTool.swift

```diff
@@ -96,25 +121,44 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
         case .listTests:
             let testPath = try buildTestsIfNeeded(options)
             let testSuites = try getTestSuites(path: testPath)
+            let tests = testSuites.filteredTests(specifier: options.testCaseSpecifier)
+
             // Print the tests.
-            for testSuite in testSuites {
-                for testCase in testSuite.tests {
-                    for test in testCase.tests {
-                        print(testCase.name + "/" + test)
-                    }
-                }
+            for test in tests {
+                print(test.specifier)
             }
 
-        case .run(let specifier):
+        case .runSerial:
             let testPath = try buildTestsIfNeeded(options)
-            let success: Bool = TestRunner(path: testPath, xctestArg: specifier, processSet: processSet).test()
-            exit(success ? 0 : 1)
+            let testSuites = try getTestSuites(path: testPath)
+            var ranSuccessfully = true
+
+            if case .none = options.testCaseSpecifier {
```

Comment:
I don't think there is need to throw `fatalError` here. All we need to do is:

1. If testCaseSpecifier is `.none`, ie., `swift test` without any filter/specifier option, don't filter any tests and run all of them serially.
2. If testCaseSpecifier is either `.regex` or `.specific`, we need to filter and run only filtered tests. In addition to that, if it is `.specific`, deprecation diagnostics need to be emitted.